admin/src/react-components/import-content.js

```diff
@@ -73,129 +80,353 @@ class ImportContentComponent extends Component {
+    this.setImportFields(url, i => (i.isEnabled = isEnabled));
+  }
+
+  setImportIsDefault(url, isDefault) {
+    this.setImportFields(url, i => (i.isDefault = isDefault));
+  }
+
+  setImportIsBase(url, isBase) {
+    this.setImportFields(url, i => (i.isBase = isBase));
+  }
+
+  setImportIsFeatured(url, isFeatured) {
+    this.setImportFields(url, i => (i.isFeatured = isFeatured));
+  }
+
+  setImportFields(url, setter) {
+    const { imports } = this.state;
+
+    for (let i = 0; i < imports.length; i++) {
+      if (imports[i].url === url) {
+        setter(imports[i]);
+        break;
+      }
+    }
+
+    this.setState({ imports });
+  }
+
+  async onPreviewImport(e) {
     if (e) e.preventDefault();
 
-    const apiInfo = this.apiInfoForSubmittedUrl(this.state.url);
+    const urls = this.state.urls.split(/[, ]+/);
+    if (!urls.find(u => u.length !== 0)) return;
 
-    if (!apiInfo) {
-      this.setState({ importResult: RESULTS.failed });
-      console.error(e);
+    const needsBaseAvatar = this.state.reticulumMeta.repo && !this.state.reticulumMeta.repo.avatar_listings.base;
+    const needsDefaultAvatar = this.state.reticulumMeta.repo && !this.state.reticulumMeta.repo.avatar_listings.default;
+    const needsDefaultScene = this.state.reticulumMeta.repo && !this.state.reticulumMeta.repo.scene_listings.default;
+
+    let hadUrl = false;
+    await new Promise(r => this.setState({ imports: [] }, r));
+    this.setState({ isLoading: true });
+
+    const importableUrls = [];
+
+    for (let i = 0; i < urls.length; i++) {
+      const url = urls[i];
+
+      if (url.endsWith(".pack")) {
+        const res = await fetch(`https://${configs.CORS_PROXY_SERVER}/${url}`);
+        const packUrls = (await res.text()).split("\n");
+        for (const u of packUrls) {
+          if (u.trim() !== "") {
+            importableUrls.push(u);
+          }
+        }
+      } else {
+        importableUrls.push(url);
+      }
     }
 
-    const { url, isScene } = apiInfo;
-    const type = isScene ? "scenes" : "avatars";
-    const columnPrefix = isScene ? "scene" : "avatar";
-    this.setState({ importing: true });
+    for (let i = 0; i < importableUrls.length; i++) {
+      const url = importableUrls[i];
+      console.log(url);
+      const apiInfo = this.apiInfoForSubmittedUrl(url);
+      if (!apiInfo) continue;
+
+      const { url: importUrl, isScene } = apiInfo;
+      const isAvatar = !isScene;
 
-    const res = await fetchReticulumAuthenticated(`/api/v1/${type}`, "POST", { url });
-    const tags = [];
+      if (!importUrl) continue;
 
-    if (this.state.addBaseTag) {
-      tags.push("base");
+      const res = await fetch(`https://${configs.CORS_PROXY_SERVER}/${importUrl}`);
+      const type = isScene ? "scenes" : "avatars";
+      const asset = (await res.json())[type][0];
+      const isDefault = (isScene && needsDefaultScene) || (isAvatar && needsDefaultAvatar);
+      const isBase = isAvatar && needsBaseAvatar && i === 0;
```

Comment:
Bit odd to condition this on `i === 0` if we might `continue` on some urls above.